### ISSUE TITLE
Renovate-On-Prem does not sign commits?

### ISSUE THREAD

**Issue by TobiX** [whitesource/renovate-on-prem#182]
I was using the Renovate OpenSource helm chart until ~05.01. and then switched to Renovate-On-Prem (using my own helm chart as a kind of combination of the sample in this repository and some parts of the OpenSource helm chart), but since I switched, Renovate has stopped signing commits. The configuration is basically unchanged: I was previously injecting the private key as a Kubernetes secret named `RENOVATE_GIT_PRIVATE_KEY`, which was used by the OpenSource helm chart to populate `gitPrivateKey`. I can see in the logs that `gitPrivateKey` is populated, but none of the debug logs from https://github.com/renovatebot/renovate/blob/main/lib/util/git/private-key.ts can be observed in the log file.

Slightly shortened config log line:
```
{"name":"renovate","hostname":"renovate-XXX","pid":8,"level":20,"logContext":"IZItLDhEfjNyfiBY5Qz8L","config":{"autodiscover":true,"dryRun":false,"gitAuthor":"Renovate Bot <XXX>","gitPrivateKey":"***********","onboardingConfigFileName":".gitlab/renovate.json","onboardingConfig":{"extends":["config:base"]},"hostRules":[XXX]},"msg":"File config","time":"2022-01-19T18:29:14.756Z","v":0}
```

Platform: GitLab (internal EE installation)

**Comment by natan-abolafya** [comment]
Hi,
I also have this problem and had created a question to the renovate repository. I'll paste the same here in case there is something useful.

============================================

I've been trying various things, read the code and so forth but I can't figure it out. It doesn't sign it or even attempts to sign it according to the debug logs.

It's renovate 2.1.0 docker image.
I pass config as environment variable. It looks like it picks up the key just fine.

```
DEBUG: Detected config in env RENOVATE_CONFIG
2022-01-21T15:03:52.313738500Z        "config": {"allowedPostUpgradeCommands": ["./gradlew .*"], "gitPrivateKey": "***********"}
...
DEBUG: Combined config
2022-01-21T15:03:52.319116100Z        "config": {
2022-01-21T15:03:52.319125000Z          "allowedPostUpgradeCommands": ["./gradlew .*"],
2022-01-21T15:03:52.319128600Z          "gitPrivateKey": "***********",
2022-01-21T15:03:52.319131600Z          "hostRules": [],
2022-01-21T15:03:52.319134600Z          "binarySource": "install",
2022-01-21T15:03:52.319137600Z          "platform": "github",
2022-01-21T15:03:52.319140600Z          "endpoint": "https://api.github.com"
2022-01-21T15:03:52.319143700Z        }
```

According to the docs, it's supposed to load it on first commit. But I don't see any attempts. The related logs (I think):
```
DEBUG: Branch needs creating (repository=appgate-sdp-int/stratus-controller, branch=renovate/com.github.jk1.dependency-license-report-2.x)
2022-01-21T15:08:48.845791800Z DEBUG: Using reuseExistingBranch: false (repository=appgate-sdp-int/stratus-controller, branch=renovate/com.github.jk1.dependency-license-report-2.x)
2022-01-21T15:08:48.846343600Z DEBUG: manager.getUpdatedPackageFiles() reuseExistinbranch=false (repository=appgate-sdp-int/stratus-controller, branch=renovate/com.github.jk1.dependency-license-report-2.x)
2022-01-21T15:08:48.952497400Z DEBUG: Updating com.github.jk1.dependency-license-report in Service.Controller/build.gradle (repository=appgate-sdp-int/stratus-controller, branch=renovate/com.github.jk1.dependency-license-report-2.x)
2022-01-21T15:08:48.952627800Z DEBUG: Updated 1 package files (repository=appgate-sdp-int/stratus-controller, branch=renovate/com.github.jk1.dependency-license-report-2.x)
2022-01-21T15:08:48.953615000Z DEBUG: No updated lock files in branch (repository=appgate-sdp-int/stratus-controller, branch=renovate/com.github.jk1.dependency-license-report-2.x)
2022-01-21T15:08:48.961230100Z DEBUG: Executing post-upgrade task (repository=appgate-sdp-int/stratus-controller, branch=renovate/com.github.jk1.dependency-license-report-2.x)

..... long postUpgradeTask stdout .....

DEBUG: Post-upgrade file saved (repository=appgate-sdp-int/stratus-controller, branch=renovate/com.github.jk1.dependency-license-report-2.x)
2022-01-21T15:10:11.956737900Z        "dep": "com.github.jk1.dependency-license-report",
2022-01-21T15:10:11.956760000Z        "file": "Service.Controller/build.gradle",
2022-01-21T15:10:11.956766200Z        "pattern": "**/**"
2022-01-21T15:10:11.967540900Z DEBUG: 1 file(s) to commit (repository=appgate-sdp-int/stratus-controller, branch=renovate/com.github.jk1.dependency-license-report-2.x)
2022-01-21T15:10:11.967837500Z DEBUG: Committing files to branch renovate/com.github.jk1.dependency-license-report-2.x (repository=appgate-sdp-int/stratus-controller, branch=renovate/com.github.jk1.dependency-license-report-2.x)
2022-01-21T15:10:11.968120200Z DEBUG: Setting git author name (repository=appgate-sdp-int/stratus-controller, branch=renovate/com.github.jk1.dependency-license-report-2.x)
2022-01-21T15:10:11.968143000Z        "gitAuthorName": "renovate-appgate-test[bot]"
2022-01-21T15:10:12.056094600Z DEBUG: Setting git author email (repository=appgate-sdp-int/stratus-controller, branch=renovate/com.github.jk1.dependency-license-report-2.x)
2022-01-21T15:10:12.056140100Z        "gitAuthorEmail": "renovate-appgate-test[bot]@users.noreply.github.com"
2022-01-21T15:10:29.132910600Z DEBUG: git commit (repository=appgate-sdp-int/stratus-controller, branch=renovate/com.github.jk1.dependency-license-report-2.x)
2022-01-21T15:10:29.132953200Z        "deletedFiles": [],
2022-01-21T15:10:29.132961400Z        "ignoredFiles": [],
2022-01-21T15:10:29.132965000Z        "result": {
2022-01-21T15:10:29.132968300Z          "author": null,
2022-01-21T15:10:29.132971400Z          "branch": "renovate/com.github.jk1.dependency-license-report-2.x",
2022-01-21T15:10:29.132974600Z          "commit": "f40de797c",
2022-01-21T15:10:29.132977600Z          "root": false,
2022-01-21T15:10:29.132980700Z          "summary": {"changes": 1, "insertions": 1, "deletions": 1}
2022-01-21T15:10:29.132983900Z        }
2022-01-21T15:10:30.972917500Z DEBUG: git push .....
```

Besides these, I searched the logs for "private", "gpg", "import" and some more relevant keywords; no mention of anything.
There was a similar topic about this which suggested the key should not have any passphrase. It was the case for me but fixing that didn't help either. Plus, I imagine there would be debug logs about the attempt to import I imagine..

Something else I should be looking at?

**Comment by rarkins** [comment]
I'm working on a fix for this

**Comment by konrad-ohms** [comment]
Thank you very much Rhys 🙂

**Comment by rarkins** [comment]
I wasn't able to test this in a real repo environment yet, but I'm optimistic enough and the fix is small enough that it was safe to release as `2.1.1` - please try

**Comment by konrad-ohms** [comment]
Great, thanks a lot, I just verified that commits are getting signed again as before.

**Comment by TobiX** [comment]
@rarkins Thanks for the quick fix, I can also confirm that commits are signed again 🎉

**Comment by rarkins** [comment]
Thanks for reporting it, and sorry for any time wasted..

**Comment by konrad-ohms** [comment]
No worries, actually the migration from OSS to on-premises worked quite smoothly, the signing part was the only problem I recognized. I thing the most time consuming part was to figure out how I get permission in my organization to create and install a new GitHub App to my repositories 😁

**Comment by TobiX** [comment]
Did this regress in 2.2.0 again?

**Comment by TobiX** [comment]
@rarkins This seems to be broken again with 2.2.0 😢

**Comment by natan-abolafya** [comment]
recently bumped from 2.1.1 to 2.4 and am having the same problem. It's perhaps due to this change https://github.com/renovatebot/renovate/issues/11537 and we need to sign it differently?

Anyone found a fix?

**Comment by rarkins** [comment]
If you're on GitHub, try setting platformCommit=true. If it works, it will mean you don't need to configure the GPG key anymore

**Comment by natan-abolafya** [comment]
that did the job. thank you!

**Comment by nekator** [comment]
so i tried all minor versions after 2.1.1 and non of them is working with gitlab :(

Any thoughts how to get gpg signing working with gitlab? Because the platformCommit workaround is github specific.

**Comment by TobiX** [comment]
For me, this is fixed in 2.6.0 (probably through #228)

**Comment by nekator** [comment]
> For me, this is fixed in 2.6.0 (probably through #228)

same here.
